Assets/Scripts/Audio/AudioManager.cs

```
@@ -0,0 +1,92 @@
+using UnityEngine;
+using FMODUnity;
+using FMOD.Studio;
+
+public enum AudioSurfaceType : int
+{
+    Concrete,
+    Wood,
+    Gravel,
+    Metal
+}
+
+public class AudioManager : MonoBehaviour
+{
+
+
+
+    public static AudioManager instance;
+
+    [SerializeField] private EventReference _testSound;
+
+    [SerializeField] private EventReference _pauseAudioEvent;
+    [SerializeField] private EventReference _pauseMenuSnapshot;
+
+    private const string _inGameSFXBusName = "bus:/SFX/In-World";
+    private const string _inGameUIBusName = "bus:/SFX/UI/In-World";
+    private const string _inGameVOBusName = "bus:/VO/In-World";
+    private const string _inGameMusicBusName = "bus:/Music/In-World";
+
+    public Bus _inGameSFXBus { get; private set; }
+    public Bus _inGameUIBus { get; private set; }
+    public Bus _inGameVOBus { get; private set; }
+    public Bus _inGameMusicBus { get; private set; }
+    private EventInstance _pauseMenuSnapshotInstance;
+
+    private void Awake()
+    {
+        instance = this;
+        _inGameSFXBus = RuntimeManager.GetBus(_inGameSFXBusName);
+        _inGameUIBus = RuntimeManager.GetBus(_inGameUIBusName);
+        _inGameVOBus = RuntimeManager.GetBus(_inGameVOBusName);
+        _inGameMusicBus = RuntimeManager.GetBus(_inGameMusicBusName);
+        _pauseMenuSnapshotInstance = RuntimeManager.CreateInstance(_pauseMenuSnapshot);
+    }
+
+
+    /// <summary>
+    /// Checks bus validity and sets it to paused
+    /// </summary>
+    public void PauseAudioBus(Bus bus)
+    {
+        if (bus.isValid())
+        {
+            bool _busPausedStatus;
+            bus.getPaused(out _busPausedStatus);
+            if (!_busPausedStatus)
+            {
+                bus.setPaused(true);
+            }
+        }
+        else
+        {
+            Debug.Log("referred bus to pause is invalid");
```

Would an error or a warning be more appropriate here? Maybe throw an exception? How expected is an invalid bus?